src/main/java/com/thealgorithms/strings/WordUpperCase.java

```diff
@@ -0,0 +1,28 @@
+package com.thealgorithms.strings;
+
+public class WordUpperCase {
+    public static void main(String[] args) {
+        String str = "hii my name is abhishek";
+        System.out.println(toUpperCase(str));
+    }
+
+    public static String toUpperCase(String str) {
+        StringBuilder sb = new StringBuilder("");
+        if (str == null || str.isEmpty()) {
+            return str; // Handle null or empty input gracefully
+        }
+        char ch = Character.toUpperCase(str.charAt(0));
+        sb.append(ch);
+        for (int i = 1; i < str.length(); i++) {
+            if (str.charAt(i) == ' ' && i < str.length() - 1) {
+                sb.append(str.charAt(i));
+                i++;
```

Comment:
The logic here is a bit hard to follow due to the manual increment of 'i'.
I think we could improve readability by using a while loop or extracting this logic into a separate method.